FreeRTOS/FreeRTOS-Plus-TCP
# Add checks to reject IP packets with broadcast source IP/MAC address

Description
-----------
This PR adds checks to filter out IP packets with broadcast/multicast source MAC/IP addresses. These kind of packets are supposed to be rejected in conformance with [RFC 1112 section 3.2.1.3](https://datatracker.ietf.org/doc/html/rfc1122#section-3.2.1.3).

Test Steps
-----------
Send IP packets with source MAC set as broadcast address (`FF:FF:FF:FF:FF:FF`) OR source IP address set as a multicast address OR both to the device running FreeRTOS+TCP stack. Currently, those packets will be accepted and processed. 
After the change mentioned in this PR, the above packet will be dropped and not processed further.

Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
